steps/hierarchical_ts/step.md

```diff
@@ -98,14 +100,98 @@ print(forecasts)
 **Hierarchical Forecast Design Sketch**
 (This will generate the coherent forecasts across the hierarchy)
 
+Added the hierarchical_encoding part which I had missed(after suggestions from **@AngelPone** and **@fkiraly**)
+
 ```
-hierarchical_reconciler = HierarchicalReconcile(method="ols")
+hierarchical_encoding=create_hierarchical_encodings(df) #Added this part, which I had forgot in the initial sketch
+hierarchical_reconciler = HierarchicalReconcile(hierarchical_encoding,method="ols")
 hierarchical_reconciler.fit(ts)
 coherent_forecasts = hierarchical_reconciler.predict(forecasts)
 print(coherent_forecasts)
 ```
 ![coherent_forecasts](e_prop_4.png)<br>
 
+## Refined API Design Sketch(as per suggestions from @AngelPone) 
+```
+class BaseHierarchy():
+    def aggregate_ts():
+    # construct df used for generating base forecasts and reconciliation
+
+class CrossSectionalHierarchy(BaseHierarchy):
+    # classmethod used for construction of hierarchy from different abstract descriptions.
+    @classmethod
+    def from_xxx():
+        return cls()
+
+class TemporalHierarchy(BaseHierarchy):
+    def __init__():
+
+# Forecasters
+def wls():
+
+def mint():
+
+class HierarchicalForecaster(BaseForecaster):
+    def __init__(hierarchy, base_forecasters:Union[BaseForecaster|List|Dict], method="ols", 
+                  weights=Optional[str])
+
+    def _check_y_X():
+
+    def _fit_forecasters(y, X, return_residuals=False):
+    def _predict_forecasters(X):
+
+    def _fit(y, X):
+        df = self.hierarchy.aggregate_ts(y)
+
```

Comment:
why fo we reimplement instead of interfacing `pyhts`?

Reply:
Your views on this @aiwalter, @ltsaprounis ??